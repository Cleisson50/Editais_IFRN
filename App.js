import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import Campus from "./src/pages/Campus";
import Index from "./src/pages/Index";
import Ano from "./src/pages/Ano";
import Edital from "./src/pages/Edital";
 
const Stack = createNativeStackNavigator();
 
export default ()  => {
  return(
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Index" screenOptions={{headerShown: true}}>
      <Stack.Screen name="Index" component={Index} options={{title:"Editais IFRN", headerStyle: {
            backgroundColor: '#4AF56F',
          },
          headerTintColor: '#fff',
          headerTitleAlign: 'center',
          headerTitleStyle: {
            fontWeight: 'bold',},}}/>
      <Stack.Screen name="Campus" component={Campus} options={{title:"Editais IFRN/ Campus", headerStyle: {
            backgroundColor: '#4AF56F',
          },
          headerTintColor: '#fff',
          headerTitleAlign: 'center',
          headerTitleStyle: {
            fontWeight: 'bold',},}} />
      <Stack.Screen name="Ano" component={Ano} options={{title:"Editais IFRN/ Campus/ Ano", headerStyle: {
            backgroundColor: '#4AF56F',
          },
          headerTintColor: '#fff',
          headerTitleAlign: 'center',
          headerTitleStyle: {
            fontWeight: 'bold',},}}/>
      <Stack.Screen name="Edital" component={Edital} options={{title:"Editais IFRN/ Campus/ Ano/ Edital", headerStyle: {
            backgroundColor: '#4AF56F',
          },
          headerTintColor: '#fff',
          headerTitleAlign: 'center',
          headerTitleStyle: {
            fontWeight: 'bold',},}}/>
    </Stack.Navigator>
      </NavigationContainer>
  )
}
