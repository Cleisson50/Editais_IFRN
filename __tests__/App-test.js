import renderer from 'react-test-renderer';
import App from '../App';
import { fireEvent, render } from '@testing-library/react-native';
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

test('teste de snapshot', () => {
    const tela = renderer.create(<App />).toJSON();
    expect(tela).toMatchSnapshot();
});

describe("Teste de Navegação", () => {
    it("Navegação de Index para Campus", () => {
        const { getByText, getByTestId } = render(<App />);

        const botao = getByTestId("botão1")
        fireEvent.press(botao)

        expect(getByText('Campus Apodi')).toBeTruthy()
    })

    it("Navegação de Index para Campus", () => {
        const { getByText, getByTestId } = render(<App />);

        const botao = getByTestId("botão2")
        fireEvent.press(botao)

        expect(getByText('Campus Apodi')).toBeTruthy()
    })

    it("Navegação de Index para Campus", () => {
        const { getByText, getByTestId } = render(<App />);

        const botao = getByTestId("botão3")
        fireEvent.press(botao)

        expect(getByText('Campus Apodi')).toBeTruthy()
    })

    it("Navegação de Index para Campus", () => {
        const { getByText, getByTestId } = render(<App />);

        const botao = getByTestId("botão4")
        fireEvent.press(botao)

        expect(getByText('Campus Apodi')).toBeTruthy()
    })
})