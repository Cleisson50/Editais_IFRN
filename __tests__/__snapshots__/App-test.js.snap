// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teste de snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="#2E8B57"
        color="#fff"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Editais IFRN"
        titleColor="#fff"
        titleFontWeight="bold"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#8FBC8F",
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#008B8B",
                "borderRadius": 30,
                "marginBottom": 20,
                "marginTop": 40,
                "opacity": 1,
                "padding": 15,
                "textAlign": "center",
              }
            }
            testID="botão1"
          >
            <View>
              <Text>
                Editais Graduação
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#008B8B",
                "borderRadius": 30,
                "marginBottom": 20,
                "opacity": 1,
                "padding": 15,
                "textAlign": "center",
              }
            }
            testID="botão2"
          >
            <View>
              <Text>
                Editais Integrado
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#008B8B",
                "borderRadius": 30,
                "marginBottom": 20,
                "opacity": 1,
                "padding": 15,
                "textAlign": "center",
              }
            }
            testID="botão3"
          >
            <View>
              <Text>
                Editais Estágio
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#008B8B",
                "borderRadius": 30,
                "marginBottom": 20,
                "opacity": 1,
                "padding": 15,
                "textAlign": "center",
              }
            }
            testID="botão4"
          >
            <View>
              <Text>
                Editais Tecnico
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
